### PR TITLE
[ADD][12.0] new module res_company_code

### PR DESCRIPTION
**Pull request by legalsylvain**
propose new trivial module that adds a new field ``code`` in the ``res.company`` model.

Code are often used in huge multi-company context to identify more easily different companies.

res_company_form

In this context, the module hide sequence, and reorder companies by code, then by name, as it is the expected behavior for the users.

res_company_tree

The switch company menu widget so displays the companies code : 

switch_company_menu

### CODE REVIEW

**Comment by legalsylvain** [comment]
Ready for review. CC :
-  @bealdav, as discussed this morning
- @quentinDupont 

regard.

**Comment by bealdav** [comment]
see https://github.com/akretion/odoo-usability/tree/12.0/company_code

not the same specifications but the same field

File: res_company_code/views/assets.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!--
+    Copyright 2016 LasLabs Inc.
+    @author Dave Lasley <dave@laslabs.com>
```
Comment (bealdav) [inline res_company_code/views/assets.xml]: Laslabs ???

File: res_company_code/models/res_company.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    ]
+
+    @api.depends('code', 'name')
+    def _compute_complete_name(self):
+        for company in self:
+            if not company.code:
+                company.complete_name = company.name
+            else:
+                company.complete_name = "%s - %s" % (
+                    company.code, company.name)
```
Comment (bealdav) [inline res_company_code/models/res_company.py:28]: I think you could add a feature in roadmap.rst : 

- add name_search() to search company by its code

File: res_company_code/views/assets.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+-->
+
+<odoo>
+    <template id="assets_backend" inherit_id="web.assets_backend">
+        <xpath expr=".">
+            <script type="application/javascript"
+                    src="/res_company_code/static/src/js/res_company_code.js"
```
Comment (quentinDupont) [inline res_company_code/views/assets.xml]: No such file :)

File: res_company_code/views/view_res_company.xml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    </record>
+
+    <record id="view_res_company_kanban" model="ir.ui.view">
+        <field name="model">res.company</field>
+        <field name="inherit_id" ref="base.view_res_company_kanban" />
+        <field name="arch" type="xml">
+            <field name="name" position="before">
+                <field name="code"/>
+            </field>
+            <xpath expr="//i[hasclass('fa', 'fa-building')]" position="before">
```
Comment (quentinDupont) [inline res_company_code/views/view_res_company.xml]: I think I prefer to see the code name after fontawesome and write the name as in multicompany choice : fa-building "C1 - Name"

Comment (quentinDupont) [reply res_company_code/views/view_res_company.xml]: Like this : image

**Review by quentinDupont — COMMENTED** [review]
Two mini points
Tested and it works fine :)

**Review by bealdav — APPROVED** [review]
Thanks, 

just a small reserve on module name `too technical` for me as I told you but that's OK

File: res_company_code/views/assets.xml (hunk 3 above)
Comment (legalsylvain) [reply res_company_code/views/assets.xml]: indeed. the whole file assets.xml is dead code. removed.

File: res_company_code/views/assets.xml (hunk 1 above)
Comment (legalsylvain) [reply res_company_code/views/assets.xml]: bad copy past. Removed.

File: res_company_code/models/res_company.py (hunk 2 above)
Comment (legalsylvain) [reply res_company_code/models/res_company.py:28]: roadmap file added.

File: res_company_code/views/view_res_company.xml (hunk 4 above)
Comment (legalsylvain) [reply res_company_code/views/view_res_company.xml]: Done ! thanks.

**Review by quentinDupont — APPROVED** [review]

**Comment by legalsylvain** [comment]
Thanks for your reviews.
/ocabot merge

**Comment by bealdav** [comment]
fixup! auto squash doesn't work :-(

https://github.com/OCA/multi-company/commits/12.0

**Comment by legalsylvain** [comment]
@bealdav : Sorry, I didn't know.

@OCA-git-bot, please the next time, could you autosquach my commits ?

Ref : https://github.com/OCA/oca-github-bot/issues/43

**Pull request merged**